### PR TITLE
Added configuration field for setting Kotlin compilers' -jdk-home flag

### PR DESCRIPTION
**Pull request by sellophane**

### CODE REVIEW

File: kotlin-eclipse-core/src/org/jetbrains/kotlin/core/compiler/KotlinCompiler.java [hunk 1]
```diff
@@ -106,7 +106,14 @@ public KotlinCompilerResult execKotlinCompiler(@NotNull String[] arguments) {
         List<String> command = new ArrayList<>();
         command.add("-kotlin-home");
         command.add(ProjectUtils.getKtHome());
-        command.add("-no-jdk");
+
+        boolean jdkHomeUndefined = kotlinProperties.isJDKHomDefined();
```
Comment (Kordyjan) [inline kotlin-eclipse-core/src/org/jetbrains/kotlin/core/compiler/KotlinCompiler.java]: I think the names here are confusing. Are those properties `true` when jdk home is defined or undefined?

File: kotlin-eclipse-core/src/org/jetbrains/kotlin/core/preferences/KotlinProperties.kt [hunk 2]
```diff
@@ -34,10 +34,14 @@ class KotlinProperties(scope: IScopeContext = InstanceScope.INSTANCE) : Preferen
         override fun writer(value: ApiVersion) = value.versionString
     }
 
+    var jdkHome by StringPreference()
+
     val compilerPlugins by ChildCollection(::CompilerPlugin)
 
     var compilerFlags by StringPreference()
 
+    fun isJDKHomDefined() = jdkHome.isNullOrBlank()
```
Comment (Kordyjan) [inline kotlin-eclipse-core/src/org/jetbrains/kotlin/core/preferences/KotlinProperties.kt]: It should be `isJDKHomeUndefined` it think.

File: kotlin-eclipse-ui/src/org/jetbrains/kotlin/preferences/views/CompilerPropertiesView.kt [hunk 3]
```diff
@@ -1,11 +1,11 @@
 package org.jetbrains.kotlin.preferences.views
 
+import javafx.beans.property.StringProperty
 import org.eclipse.jface.resource.FontDescriptor
 import org.eclipse.swt.SWT
-import org.eclipse.swt.widgets.Button
-import org.eclipse.swt.widgets.Composite
-import org.eclipse.swt.widgets.Display
-import org.eclipse.swt.widgets.Label
+import org.eclipse.swt.graphics.Color
```
Comment (Kordyjan) [inline kotlin-eclipse-ui/src/org/jetbrains/kotlin/preferences/views/CompilerPropertiesView.kt]: It is leftover, isn't it?

**Comment by zarechenskiy** [comment]
If I'm right, previously it was possible to change JDK via Java project configuration, now it's also possible to define JDK home via Kotlin compiler settings, is it right? So now we have two sources of this information and can't it be dangerous when Java is compiled with one version and Kotlin with another?

**Pull request merged**